src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.Web/Program.Aspire.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.VectorData;
+using ChatWithCustomData_CSharp.Web.Components;
+using ChatWithCustomData_CSharp.Web.Services;
+using ChatWithCustomData_CSharp.Web.Services.Ingestion;
+#if (IsOllama)
+#elif (IsGHModels)
+using OpenAI;
+using System.ClientModel;
+#else // IsAzureOpenAI || IsOpenAI
+using OpenAI;
+#endif
+#if (UseAzureAISearch)
+using Microsoft.SemanticKernel.Connectors.AzureAISearch;
+#elif (UseQdrant)
+using Microsoft.SemanticKernel.Connectors.Qdrant;
+#endif
+
+var builder = WebApplication.CreateBuilder(args);
+builder.AddServiceDefaults();
+builder.Services.AddRazorComponents().AddInteractiveServerComponents();
+
+#if (IsGHModels)
+var credential = new ApiKeyCredential(builder.Configuration["GITHUB_MODELS_TOKEN"] ?? throw new InvalidOperationException("Missing configuration: GITHUB_MODELS_TOKEN. See the README for details."));
+var openAIOptions = new OpenAIClientOptions()
+{
+    Endpoint = new Uri("https://models.inference.ai.azure.com")
+};
+
+var ghModelsClient = new OpenAIClient(credential, openAIOptions);
+var chatClient = ghModelsClient.AsChatClient("gpt-4o-mini");
+var embeddingGenerator = ghModelsClient.AsEmbeddingGenerator("text-embedding-3-small");
+#elif (IsOllama)
+builder.AddOllamaApiClient("chat")
+    .AddChatClient()
+    .UseFunctionInvocation()
+    .UseLogging();
+builder.AddOllamaApiClient("embeddings")
+    .AddEmbeddingGenerator();
+#elif (IsAzureAiFoundry)
+
+#else // IsAzureOpenAI || IsOpenAI
+builder.AddOpenAIClientFromConfiguration("openai");
+#endif
+
+#if (UseAzureAISearch)
+builder.AddAzureSearchClient("azureAISearch");
+
+builder.Services.AddSingleton<IVectorStore, AzureAISearchVectorStore>();
+#elif (UseQdrant)
+builder.AddQdrantClient("vectordb");
+
+builder.Services.AddSingleton<IVectorStore, QdrantVectorStore>();
+#else // UseLocalVectorStore
+var vectorStore = new JsonVectorStore(Path.Combine(AppContext.BaseDirectory, "vector-store"));
+builder.Services.AddSingleton<IVectorStore>(vectorStore);
+#endif
+builder.Services.AddScoped<DataIngestor>();
+builder.Services.AddSingleton<SemanticSearch>();
+#if (IsOllama)
+#elif (IsGHModels)
+builder.Services.AddChatClient(chatClient).UseFunctionInvocation().UseLogging();
+builder.Services.AddEmbeddingGenerator(embeddingGenerator);
+#else // IsAzureOpenAI || IsOpenAI
+builder.Services.AddChatClient(sp => sp.GetRequiredService<OpenAIClient>().AsChatClient("gpt-4o-mini"))
+    .UseFunctionInvocation()
+    .UseLogging();
```

Comment:
If we switch to use the Aspire package for this, otel will be enabled automatically. But if we don't then we should add:

```cs
    .UseOpenTelemetry();
```

The equivalent is true for L62 as well.